samza-core/src/main/java/org/apache/samza/runtime/RemoteApplicationRunner.java

```diff
@@ -46,25 +53,91 @@ public RemoteApplicationRunner(Config config) {
    * Run the {@link StreamApplication} on the remote cluster
    * @param app a StreamApplication
    */
-  @Override public void run(StreamApplication app) {
+  @Override
+  public void run(StreamApplication app) {
     try {
-      // 1. build stream graph
-      StreamGraph streamGraph = new StreamGraphImpl(this, config);
-      app.init(streamGraph, config);
+      // 1. initialize and plan
+      ProcessorGraph processorGraph = getExecutionPlan(app);
 
-      // 2. create the physical execution plan
-      ExecutionPlanner planner = new ExecutionPlanner(config);
-      ProcessorGraph processorGraph = planner.plan(streamGraph);
+      // 2. create the necessary streams
+      List<StreamSpec> streams = processorGraph.getIntermediateStreams().stream()
+          .map(streamEdge -> streamEdge.getStreamSpec())
+          .collect(Collectors.toList());
+      StreamManager streamManager = new StreamManager(new JavaSystemConfig(config).getSystemAdmins());
```

Comment:
Can we create StreamManager in the beginning of run() and pass it into the planner so the system admins won't be created twice?